### PR TITLE
PRIME-1927 Update Email Status doesnt priority non-statuses

### PR DESCRIPTION
**Pull request by johnnyl6**
- Added `UpdateCount` column to track number of update from CHES
- Added `OrderBy UpdateCount` in query to lower priority of email with more `UpdateCount`

### CODE REVIEW

File: prime-dotnet-webapi/Services/EmailService.cs [hunk 1]
```diff
@@ -310,18 +310,24 @@ public async Task<int> UpdateEmailLogStatuses(int limit)
 
             var emailLogs = await _context.EmailLogs
                 .Where(predicate)
+                .Where(e => e.LatestStatus != ChesStatus.Pending)
```
Comment (james-hollinger) [inline prime-dotnet-webapi/Services/EmailService.cs:313]: so this will never update email logs that are pending?

**Review by james-hollinger — CHANGES_REQUESTED** [review]
There was a discussion about this ticket when it was first intook a sprint or two ago and it looks like those details weren’t passed on and they weren’t outlined in the ticket appropriately. We should have a quick talk about it since at the time we assumed someone else was going to make these updates with the knowledge of those discussions.

File: prime-dotnet-webapi/Models/Email/EmailLog.cs [hunk 2]
```diff
@@ -29,6 +29,8 @@ public class EmailLog : BaseAuditable
 
         public string StatusMessage { get; set; }
 
+        public int? UpdateCount { get; set; }
```
Comment (james-hollinger) [inline prime-dotnet-webapi/Models/Email/EmailLog.cs:32]: `UpdateCount` should be an `int`, which will initialize at 0

File: prime-dotnet-webapi/Services/EmailService.cs [hunk 3]
```diff
@@ -310,6 +310,7 @@ public async Task<int> UpdateEmailLogStatuses(int limit)
 
             var emailLogs = await _context.EmailLogs
                 .Where(predicate)
+                .OrderBy(e => e.UpdateCount)
                 .OrderBy(e => e.UpdatedTimeStamp)
```
Comment (james-hollinger) [inline prime-dotnet-webapi/Services/EmailService.cs:314]: you need to use `.ThenBy(e => e.UpdatedTimeStamp)` here

File: prime-dotnet-webapi/Services/EmailService.cs [hunk 4]
```diff
@@ -321,6 +322,8 @@ public async Task<int> UpdateEmailLogStatuses(int limit)
                 {
                     email.LatestStatus = status;
                 }
+                email.UpdateCount = email.UpdateCount.HasValue ? email.UpdateCount.Value + 1 : 1;
```
Comment (james-hollinger) [inline prime-dotnet-webapi/Services/EmailService.cs:325]: with `UpdateCount` being an `int`, this can just be `email.UpdateCount++`

**Review by AMiKaH — APPROVED** [review]

**Review by james-hollinger — APPROVED** [review]
LGTM

**Pull request merged**